mlir/lib/Dialect/Rock/Tuning/GridwiseGemmParams.cpp

```diff
@@ -376,38 +366,14 @@ PopulateParamsXDL::isValidBlockwiseGemmXDLOPS(const InitParamsXDL &param,
   if ((param.gemmNPerBlock % param.gemmNPerWave) != 0)
     return failure();
 
-  // TODO Remove. Note KPerBlock and KPack are independent tuning parameters.
-  // There's no need to check if they are divide exactly
-  if ((param.gemmKPerBlock % param.gemmKPack) != 0)
-    return failure();
-
   // Reject invalid KPACK values.
-  // For fp32: reject anything wider than 4.
-  // For fp16/bf16: reject anything narrower than 4, or greater than 8.
-  if (dataType.isF32() && param.gemmKPack > 4) {
-    LLVM_DEBUG(llvm::dbgs() << "Invalid KPACK tuning parameter: "
-                            << param.gemmKPack << "\n");
-    return failure();
-  }
-  if ((dataType.isF16() || dataType.isBF16()) && (param.gemmKPack != 1) &&
-      ((param.gemmKPack < 4) || (param.gemmKPack > 8))) {
-    LLVM_DEBUG(llvm::dbgs() << "Invalid KPACK tuning parameter: "
-                            << param.gemmKPack << "\n");
-    return failure();
-  }
```

Comment:
We may want to remember the rational of this and file ticket to do some performance studies.

The reason why this restriction is here may be due to two reasons:
 - Implicitly there may be a coupling in global load code that just load whatever kpack given to it in one go. I think you have fixed that
 - Implicitly maybe when the kpack is too wide for a type then tuning may just yield sub-optimal solutions and there's no need to keep trying those that are too wide.

So I'm more referring to the second bullet point (performance) than the 1st (correctness).

Reply:
I figured these were here for correctness which is why I replaced them by the `isValid` check

Reply:
And yeah, once Yi's tool happens, I'll be happy to do perf studies with it